backend/src/main/java/hu/flowacademy/vajdasagbrand/service/ItemService.java

```diff
@@ -4,77 +4,98 @@
         }
-        if (!StringUtils.hasText(item.getBio())) {
+        if(!StringUtils.hasText(item.getBio())){
             throw new ValidationException("Didn't get bio");
         }
-        if (item.getScore() < 0) {
+        if(item.getScore() == 0 || item.getScore() > 100) {
             throw new ValidationException("Impossible value");
         }
-        if (!StringUtils.hasText(item.getAddress())) {
+        if(!StringUtils.hasText(item.getAddress())){
             throw new ValidationException("Didn't get address");
         }
-        if (!StringUtils.hasText(item.getCity())) {
+        if(!StringUtils.hasText(item.getCity())){
             throw new ValidationException("Didn't get city");
         }
-        if (item.getCategory() == null) {
+        if(item.getCategory() == null) {
             throw new ValidationException("Didn't get category");
         }
-        if (!StringUtils.hasText(item.getCoordinateX())) {
+        if(!StringUtils.hasText(item.getCoordinateX())) {
             throw new ValidationException("Didn't get coordinate_x");
         }
-        if (!StringUtils.hasText(item.getCoordinateY())) {
+        if(!StringUtils.hasText(item.getCoordinateY())) {
             throw new ValidationException("Didn't get coordinate_y");
         }
-        if (!StringUtils.hasText(item.getPhone())) {
+        if(!StringUtils.hasText(item.getPhone())) {
             throw new ValidationException("Didn't get phone");
         }
-        if (!StringUtils.hasText(item.getWebsite())) {
+        if(!StringUtils.hasText(item.getWebsite())) {
             throw new ValidationException("Didn't get website");
         }
-        if (!StringUtils.hasText(item.getFacebook())) {
+        if(!StringUtils.hasText(item.getFacebook())) {
             throw new ValidationException("Didn't get facebook");
         }
-        if (!StringUtils.hasText(item.getInstagram())) {
+        if(!StringUtils.hasText(item.getInstagram())) {
             throw new ValidationException("Didn't get instagram");
         }
     }
+
+    private void modifyItems(Item item, Item tempItem) {
+        tempItem.setName(item.getName());
+        tempItem.setBio(item.getBio());
+        tempItem.setScore(item.getScore());
+        tempItem.setAddress(item.getAddress());
+        tempItem.setCity(item.getCity());
+        tempItem.setCategory(item.getCategory());
+        tempItem.setCoordinateX(item.getCoordinateX());
+        tempItem.setCoordinateY(item.getCoordinateY());
+        tempItem.setPhone(item.getPhone());
+        tempItem.setWebsite(item.getWebsite());
+        tempItem.setFacebook(item.getFacebook());
+        tempItem.setInstagram(item.getInstagram());
+    }
+
+    private Item findFirstByIdAndDeletedAtNotNull(String id) throws ValidationException {
+        Optional<Item> toBeDeleted = itemRepository.findById(id);
+        if (toBeDeleted.orElseThrow(() -> new ValidationException("No item found with given id")).getDeletedAt() != null) {
```

Comment:
Na ennek így se füle se farka... Az orElseThrow már alapból Item-mel tér vissza, szóval retrunölhetsz vele az if-re nincs szükség. A kivételt pedig úgyis dobja, ha nem volt elem, nem kell kétszer (nem is lehet).

Reply:
kijavítottam